File: SocialAppFlutter/social_app/lib/services/events/get_new_requested_location_info_for_payment_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/auth/basic_response.dart';
import 'package:social_app/Models/coordinates.dart';
import 'package:social_app/Models/sport_category.dart';

part 'json_generated_code/get_new_requested_location_info_for_payment_response.g.dart';

@JsonSerializable(explicitToJson: true)
class GetNewRequestedLocationInfoForPaymentResponse extends BasicResponse {
  final SportCategory? sportCategory;
  final Coordinates? coordinates;
  final String city;
  final String locationName;
  final int? approvedLocationId;
  final String ownerEmail;
  final bool found;
  final bool alreadyUsed;

  GetNewRequestedLocationInfoForPaymentResponse(
      super.statusCode,
      super.message,
      super.type,
      this.sportCategory,
      this.coordinates,
      this.city,
      this.locationName,
      this.approvedLocationId,
      this.found, this.ownerEmail, this.alreadyUsed);

  factory GetNewRequestedLocationInfoForPaymentResponse.fromJson(
          Map<String, dynamic> json) =>
      _$GetNewRequestedLocationInfoForPaymentResponseFromJson(json);

  Map<String, dynamic> toJson() =>
      _$GetNewRequestedLocationInfoForPaymentResponseToJson(this);
}

File: APIs/EventsService/EventsService/DTOs/Response/GetNewRequestedLocationInfoForPaymentResponse.cs
namespace EventsService.DTOs.Response
{
    public class GetNewRequestedLocationInfoForPaymentResponse: BasicResponse
    {
        public SportCategoryObject SportCategory { get; set; } = null!;
        public CoordinatesObject Coordinates { get; set; } = null!;
        public string City { get; set; } = string.Empty;
        public string LocationName { get; set; } = string.Empty;
        public int ApprovedLocationId { get; set; }
        public string OwnerEmail { get; set; } = string.Empty;
        public bool Found { get; set; }
        public bool AlreadyUsed { get; set; }
    }
}
